src/ResourceManager/Network/Commands.Network/NetworkWatcher/SetAzureNetworkWatcherConfigFlowLogCommand.cs

```diff
@@ -91,13 +131,93 @@ public class SetAzureNetworkWatcherConfigFlowLogCommand : NetworkWatcherBaseCmdl
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByNameWithTAByResource)]
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Flag to enable/disable retention.",
+            ParameterSetName = SetFlowlogByNameWithTAByDetails)]
+        [ValidateNotNull]
+        public SwitchParameter EnableTrafficAnalytics { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Subscription of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByResourceWithTAByDetails)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Subscription of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByNameWithTAByDetails)]
+        [ValidateNotNullOrEmpty]
+        public string WorkspaceResourceId { get; set; }
+
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "GUID of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByResourceWithTAByDetails)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "GUID of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByNameWithTAByDetails)]
+        [ValidateNotNullOrEmpty]
+        public string WorkspaceGUId { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure Region of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByResourceWithTAByDetails)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Azure Region of the WS which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByNameWithTAByDetails)]
+        [ValidateNotNullOrEmpty]
+        public string WorkspaceLocation { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The WS object which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByResourceWithTAByResource)]
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The WS object which is used to store the traffic analytics data.",
+            ParameterSetName = SetFlowlogByNameWithTAByResource)]
+        [ValidateNotNull]
+        public dynamic Workspace { get; set; }
```

Comment:
what is "dynamic"? - please use the actual type of "Workspace"

Reply:
Updating. Will addess any issues cause by this once usage of PSWorkspace type is finalized